fms/models/gpt_bigcode.py

```diff
@@ -115,20 +116,28 @@ def forward(
 
 
 class GPTBigCodeHeadless(nn.Module):
-    def __init__(self, config: GPTBigCodeConfig):
+    def __init__(
+        self, config: GPTBigCodeConfig, distributed_strategy: DistributedStrategy
+    ):
         super().__init__()
         self.config = config
+        self.distributed_strategy = distributed_strategy
 
-        self.layers = nn.ModuleList(
-            [GPTBigCodeBlock(self.config) for _ in range(self.config.nlayers)]
-        )
+        layers = []
+        for i in range(self.config.nlayers):
+            block = GPTBigCodeBlock(self.config)
+            block_module = self.distributed_strategy.distribute_layer(block, i)
+            layers.append(block_module)
+        self.layers = nn.ModuleList(layers)
 
         self.embedding = nn.Embedding(self.config.src_vocab_size, self.config.emb_dim)
```

Comment:
llama has `self.shared = self.distributed_strategy.distribute_module(shared)` where shared is WordEmbedding, is anything similar needed either here or on the head?

Reply:
also these are absolute position embeddings right? in llama Antoni had some special handling for rotary emb under uniform MP strategy but probably not applicable here

Reply:
This was related to the comment above. We need to figure out how to get this working when we tie the embeddings/heads (If you do the above, there will be shape issues when running the head). Did not want to hold up other things while we figure that out as providing tp support for GPTBigCode layers is still a major jump forward.